dask/sizeof.py

```diff
@@ -256,13 +256,20 @@ def sizeof_pyarrow_chunked_array(data):
 def _register_entry_point_plugins():
     """Register sizeof implementations exposed by the entry_point mechanism."""
     for entry_point in entry_points(group="dask.sizeof"):
-        registrar = entry_point.load()
-        try:
-            registrar(sizeof)
-        except Exception:
-            logger.exception(
-                f"Failed to register sizeof entry point {entry_point.name}"
-            )
+        # should be same as entry_point.name
+        toplevel = entry_point.name
+
+        @sizeof.register_lazy(toplevel)
+        def register_it(ep=entry_point, top=toplevel):
+            registrar = ep.load()
+            try:
+                registrar(sizeof)
+                if top in sizeof._lazy:
+                    # registrar also did a lazy register
+                    sizeof._lazy[top]()
+                    sizeof._lazy.pop(top, None)
```

Comment:
It's not clear to me what's going on here. Why is this needed? 

It also feels a bit off accessing `._lazy` directly

Reply:
The test and previous example suggested using lazy registration via entrypoints. That means that previous to this PR, the registering function would be imported, but not run. We could change the recommendation now to make an immediate registration, since it won't execute until needed any more.

Now, when doing the sizeof, the registering function is imported and run. If its effect is to make a lazy registration, running it will merely populate _lazy for next time. Dispath.dispath does the same thing when processing a lazy registration the first time. The alternative way to do it, without _ access, would be to construct/find any class defined to be in the target namespace. Not sure how to do that!